azurerm/resource_arm_kubernetes_cluster.go

```diff
@@ -39,6 +39,12 @@ func resourceArmKubernetesCluster() *schema.Resource {
 					return nil
 				}
 
+				podCidr := profile["pod_cidr"].(string)
+
+				if networkPlugin == "azure" && podCidr != "" {
+					return fmt.Errorf("`network_profile.pod_cidr` can not be used when `network_profile.network_plugin` is set to `azure`. Please remove `network_profile.pod_cidr` or set `network_profile.network_plugin` to `kubenet`.")
```

Comment:
🤔 since this is a list containing a single item, would this be clearer by either making:

`network_profile.pod_cidr` -> `network_profile.0.pod_cidr`
`network_profile.network_plugin` -> `network_profile.0.network_plugin`

or by updating this to:

```
the `pod_cidr` field in the `network_profile` block can not be specified when `network_plugin` is set to `azure`. Please remove `pod_cidr` or set `network_plugin` to `kubenet`.
```

Reply:
Hmm, I don't know about adding the index (mostly because when creating the terraform you don't really worry about the fact that it's a list so it makes it a little unclear)

I prefer your second option over all three. Thanks Tom!